src/main/java/by/exadel/internship/service/impl/InterviewServiceImpl.java

```diff
@@ -0,0 +1,156 @@
+
+            formFullDTO.setFormStatus(FormStatus.ADMIN_INTERVIEW_ASSIGNED);
+            formService.updateForm(formFullDTO);
+        }else {
+            if (formFullDTO.getFormStatus() != FormStatus.REGISTERED) {
+                throw new InappropriateRoleException("Form with uuid = " + formFullDTO.getId() + " doesn't has need Status");
+            }
+            if (userDTO.getUserRole()!=UserRole.ADMIN){
+                throw new InappropriateRoleException("User with uuid = " + userDTO.getId() + " doesn't has need Status");
+
+            }
+        }
+    }
+
+    private void deleteTime(LocalDateTime localDateTime, UUID userId) {
+        UserDTO userDTO = userService.getById(userId);
+        List<UserTimeSlotDTO> userTimeSlotDTOList = userDTO.getUserTimeSlots()
+                .stream()
+                .filter(userTimeSlotDTO -> userTimeSlotDTO.getStartDate()
+                        .equals(localDateTime)).collect(Collectors.toList());
+        userTimeSlotService.deletedById(userTimeSlotDTOList.get(0).getId());
+    }
+
+    @Override
+    public void rewriteInterviewTime(UUID formId, InterviewSaveDTO interviewDTO) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Try to save Form with Interview");
+        FormFullDTO formFullDTO = formService.getById(formId);
+        InterviewDTO interviewFullDTO = formFullDTO.getInterview();
+        UserDTO userDTO = userService.getById(interviewDTO.getUserId());
+        if(formFullDTO.getFormStatus() == FormStatus.ADMIN_INTERVIEW_ASSIGNED
+                && userDTO.getUserRole() == UserRole.ADMIN){
+            this.restoreTime(interviewFullDTO.getAdminInterviewDate(), interviewFullDTO.getAdmin());
+
+            interviewFullDTO.setAdmin(interviewDTO.getUserId());
+            interviewFullDTO.setAdminInterviewDate(interviewDTO.getUserInterviewDate());
+
+            formFullDTO.setInterview(interviewFullDTO);
+            formService.updateForm(formFullDTO);
+
+            this.deleteTime(interviewDTO.getUserInterviewDate(),interviewDTO.getUserId());
+            return;
+        }
+        if (formFullDTO.getFormStatus() == FormStatus.ADMIN_INTERVIEW_PASSED
+                && userDTO.getUserRole() == UserRole.TECH_EXPERT){
+            interviewFullDTO.setTechSpecialist(interviewDTO.getUserId());
+            interviewFullDTO.setTechInterviewDate(interviewDTO.getUserInterviewDate());
+
+            formFullDTO.setInterview(interviewFullDTO);
+            formFullDTO.setFormStatus(FormStatus.TECH_INTERVIEW_ASSIGNED);
+            formService.updateForm(formFullDTO);
+
+            this.deleteTime(interviewDTO.getUserInterviewDate(),interviewDTO.getUserId());
+            return;
+        }
+        if (formFullDTO.getFormStatus() == FormStatus.TECH_INTERVIEW_ASSIGNED
+                && userDTO.getUserRole() == UserRole.TECH_EXPERT){
+            this.restoreTime(interviewFullDTO.getTechInterviewDate(), interviewFullDTO.getTechSpecialist());
```

Comment:
combine 2 `if` into one and do `this.restoreTime` in case `TECH_INTERVIEW_ASSIGNED`

Reply:
Done